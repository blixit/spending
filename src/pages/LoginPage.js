import React from 'react';
import styled from 'styled-components';

import PageUI from 'pages/Page';

import LoginForm from 'components/forms/login-form/LoginForm';

const Page = styled(PageUI)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const LoginPage = props => {
  const { router } = props;
  return (
    <Page {...props} >
      <LoginForm router={router} ></LoginForm>
    </Page>
  )
};

LoginPage.displayName = 'LoginPage';

export default LoginPage;
