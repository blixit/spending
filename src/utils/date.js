/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};