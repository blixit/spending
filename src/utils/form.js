/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
