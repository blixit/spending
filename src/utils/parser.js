export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);