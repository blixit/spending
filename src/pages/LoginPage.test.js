import 'jest-localstorage-mock';

import { withTheme } from 'ui/theming/test.utils';

import render from 'testing/render';

import { LoginPage } from './LoginPage';

describe('[pages] LoginPage', () => {
  it('should render properly', () => {
    const tree = render(withTheme(LoginPage));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
