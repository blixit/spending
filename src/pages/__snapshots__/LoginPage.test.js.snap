// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] LoginPage should render properly 1`] = `
<div
  className="sc-bdVaJa kjZkLq sc-gqjmRU biifNA"
>
  <div
    className="sc-bwzfXH gouoTZ"
  >
    <div
      className="sc-gZMcBi llcwPd"
    >
      <h1
        className="sc-dnqmqq leIyVc"
      >
        Your workspace
      </h1>
      <div
        className="sc-gzVnrw dnKKOo"
        placeholder="Enter your login"
        value={null}
      >
        <label
          className="sc-EHOje fnacYO"
        >
          <span>
            Login
          </span>
          <input
            className="sc-ifAKCX jnltQ"
            onChange={[Function]}
            placeholder="Enter your login"
            type="text"
            value={null}
          />
        </label>
      </div>
      <div
        className="sc-gzVnrw dnKKOo"
        placeholder="Enter your password"
        value={null}
      >
        <label
          className="sc-EHOje fnacYO"
        >
          <span>
            Password
          </span>
          <input
            className="sc-ifAKCX jnltQ"
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
            value={null}
          />
        </label>
      </div>
      <div
        className="sc-iwsKbI kSxtwv"
      >
        <button
          className="sc-htpNat kqiJpG sc-bxivhb iXvLMe"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;
